src/useTranslation.js

```diff
@@ -40,11 +40,9 @@ export function useTranslation(ns, props = {}) {
 
   // binding t function to namespace (acts also as rerender trigger)
   function getT() {
-    return {
-      t: i18n.getFixedT(null, i18nOptions.nsMode === 'fallback' ? namespaces : namespaces[0]),
-    };
+    return i18n.getFixedT(null, i18nOptions.nsMode === 'fallback' ? namespaces : namespaces[0]);
   }
-  const [t, setT] = useState(getT()); // seems we can't have functions as value -> wrap it in obj
```

Comment:
what was the reason for this comment?

Reply:
I assume the original author of the code just tried `useState(getT())` with `getT()` returning the i18n.t function. React detected that the parameter passed to useState was a function and thought it was a lazy-init function (`useState(() => initialValue)`). So it was calling i18n.t with no args and used the return value as initial value, which was probably null or undefined.

Basically to store a function in a state you have to have a function which returns the function, e.g. `useState(() => (args) => console.log(args))` and `setState(() => (args) => console.log(args))`